rust/main/chains/hyperlane-sealevel/src/transaction.rs

```diff
@@ -43,38 +51,43 @@ pub fn search_dispatched_message_transactions(
                 .map(|(hash, account_keys, instructions)| (index, hash, account_keys, instructions))
         })
         .filter_map(|(index, hash, account_keys, instructions)| {
-            filter_not_relevant(
+            filter_by_relevancy(
                 mailbox_program_id,
                 message_storage_pda_pubkey,
                 hash,
                 account_keys,
                 instructions,
+                is_specified_message_instruction,
             )
             .map(|hash| (index, hash))
         })
         .collect::<Vec<(usize, H512)>>()
 }
 
-fn filter_not_relevant(
+fn filter_by_relevancy<F>(
     mailbox_program_id: &Pubkey,
     message_storage_pda_pubkey: &Pubkey,
     hash: H512,
     account_keys: Vec<String>,
     instructions: Vec<UiCompiledInstruction>,
-) -> Option<H512> {
+    is_specified_message_instruction: &F,
```

Comment:
do you expect this predicate will get more complex / abstract than just a `matches(..., Instruction(...))`? If not, you could pass an `Instruction` type here to match on below

Reply:
The predicate will be more generalised to support IGP. It will come in next PR.